drf/pkg/permissions.py

```diff
@@ -0,0 +1,82 @@
+from rest_framework import permissions
+from accounts.models import Profile
+from pkg.models import Role, Permission, RolePermission
+from django.db.models import Q
+from university.models import University, School, UniversitySchools
+
+ADD = 1
+EDIT = 2
+DELETE = 3
+
+
+class CanAddObject(permissions.BasePermission):
+    """
+    Permission for create or read university
+    """
+    message = "Adding university objects not allowed"
+
+    def has_permission(self, request, view):
+
+        if request.method in permissions.SAFE_METHODS:
+            return True
+
+        """
+        TO DO
+        Need to check how to pass queryset 
+        from UniversitySchools to has_permission arg "view"
+        This is temporary solution
+        """
+        role_type = None
+
+        if view.queryset is not None:
```

Comment:
Maybe you can create helper method which will accept one param: method_name and return type